account_analytic_distribution_manual_date/tests/test_manual_distribution_date.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2024 (APSL - Nagarro) Bernat Obrador
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from datetime import date
+
+from freezegun import freeze_time
+
+from odoo.exceptions import UserError
+from odoo.tests import TransactionCase
+
+
+class TestManualDistributionDate(TransactionCase):
+    @classmethod
+    @freeze_time("2024-01-01")
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.manual_distribution = cls.env[
+            "account.analytic.distribution.manual"
+        ].create(
+            {
+                "name": "Test",
+                "start_date": "2024-01-01",
+                "end_date": "2024-12-31",
+            }
+        )
+
+        cls.invoice = cls.env["account.move"].create(
+            {
+                "move_type": "out_invoice",
+                "invoice_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": "Product A",
+                            "quantity": 1,
+                            "price_unit": 100.0,
+                            "manual_distribution_id": cls.manual_distribution.id,
+                        },
+                    )
+                ],
```

Comment:
Could you use the [Command](https://github.com/odoo/odoo/blob/370de95190cf07f17281f0ed6365b6806663cac2/odoo/fields.py#L4227-L4256) class? It is more readable.
```suggestion
                "invoice_line_ids": [
                   Command.create(
                        {
                            "name": "Product A",
                            "quantity": 1,
                            "price_unit": 100.0,
                            "manual_distribution_id": cls.manual_distribution.id,
                        },
                    )
                ],
```